scripts/install.sh

```diff
@@ -41,44 +49,16 @@ download_binary() {
             ;;
     esac
 
-    ARG_ARCH=${2:-"$_arch"}
-
-    ARG_OUT_FILE=${3:-"./router"}
-
-    GITHUB_REPO="https://github.com/apollographql/router"
-
-    # Validate token.
-    curl -o /dev/null -s $GITHUB_REPO || { echo "Error: Invalid repo, token or network issue!";  exit 1; }
-
-    #_tardir="router-$DOWNLOAD_VERSION-${_arch}"
-    #_url="$BINARY_DOWNLOAD_PREFIX/$DOWNLOAD_VERSION/${_tardir}.tar.gz"
+    _tardir="router-$DOWNLOAD_VERSION-${_arch}"
+    _url="$BINARY_DOWNLOAD_PREFIX/$DOWNLOAD_VERSION/${_tardir}.tar.gz"
     _dir="$(mktemp -d 2>/dev/null || ensure mktemp -d -t router)"
     _file="$_dir/input.tar.gz"
     _router="$_dir/router$_ext"
 
-    _release_download_url="$GITHUB_REPO/releases"
-    _router_version=$DOWNLOAD_VERSION
-    if [ "$DOWNLOAD_VERSION" = "latest" ]; then
-        _response=$(curl -Ls -o /dev/null -w '%{url_effective}' $GITHUB_REPO/releases/latest)
-        _router_version=$(echo "$_response" | cut -d'/' -f 8)
-        [ "$_router_version" ] || { echo "Error: Failed to get asset version for '$ARG_ARCH', response: $_response" | awk 'length($0)<100' >&2; exit 1; }
-    fi;
-
-    say "Downloading release info for '$_release_download_url'"
-
-    # Cut the 'v' prefix
-    _name="router-$(echo "$_router_version" | cut -c2-)-$ARG_ARCH.tar.gz"
-
-    _url="$GITHUB_REPO/releases/download/$_router_version/$_name"
-
-    say "Found $_name" 1>&2
+    say "Downloading router from $_url ..." 1>&2
 
     ensure mkdir -p "$_dir"
-
-    # Download asset file.
-    say "Downloading router from $_url"
-    
-    curl -sSfL -H 'Accept: application/octet-stream' "$_url" -o "$_file"
+    downloader "$_url" "$_file"
     if [ $? != 0 ]; then
```

Comment:
NIT: this is considered to be better style (SC2181)
```suggestion
if ! downloader "$_url" "$_file"; then
```